streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1111,13 +1111,21 @@ RecordCollector recordCollector() {
         return recordCollector;
     }
 
+    public long streamTime() {
+        return partitionGroup.streamTime();
+    }
+
+    public long currentSystemTimeMs() {
+        return time.milliseconds(); 
+    }
+
     // below are visible for testing only
     int numBuffered() {
         return partitionGroup.numBuffered();
     }
 
-    long streamTime() {
-        return partitionGroup.streamTime();
+    void setTime(final Time time) {
```

Comment:
I don't think we need this method (cf. my other comment in the test code where it is used).